Mathlib/Topology/Algebra/InfiniteSum/Basic.lean

```diff
@@ -1189,6 +1189,36 @@ theorem Summable.subtype (hf : Summable f) (s : Set β) : Summable (f ∘ (↑)
   hf.comp_injective Subtype.coe_injective
 #align summable.subtype Summable.subtype
 
+theorem summable_iff_vanishing_tsum : Summable f ↔
```

Comment:
The direct implication in your theorem does require completeness, right? And in general the right hand side is equivalent to the fact that the sum over finsets is a Cauchy sequence, if I understand correctly. Could you give a statement like this, modelled on `cauchySeq_finset_iff_vanishing`, and then deduce your result in the complete case? Same thing for next theorem.

Reply:
You mean it *doesn't* require completeness, right? I can indeed remove the completeness assumption in the forward implications, even UniformAddGroup; only TopologicalAddGroup is required, and I've generalized [tendsto_tsum_compl_atTop_zero](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/InfiniteSum/Basic.html#tendsto_tsum_compl_atTop_zero) to remove uniformity and golfed it using my theorem. However, I don't think I can state my theorems using cauchySeq, or even fit the two theorems into the filter framework, since I'm dealing with tsums over general Sets (not necessarily Finsets or complement of Finsets); we certainly can't use `atTop` on `Set α` and have to introduce some other filter.

Reply:
I wasn't clear, sorry. I am asking if the following is true:
```
theorem foobar {f : α → G} : (CauchySeq fun s : Finset β => ∑ b in s, f b) ↔
    ∀ e ∈ 𝓝 (0 : G), ∃ s : Finset α, ∀ t : Set α, Disjoint t s → (∑' a : t, f a) ∈ e
```

Reply:
Sorry for the misunderstanding! Your approach indeed looks much nicer.